### PR TITLE
Added searchbox placeholder and pod/deployment filters

### LINKED ISSUE #25: IntelliJ target selection don't show deployment targets

### PR DESCRIPTION
**Pull request by Razz4780**
Closes #25
[Screencast from 26.06.2023 14:22:50.webm](https://github.com/metalbear-co/mirrord-intellij/assets/34063647/6c098e08-e89c-4204-9aa4-baa3f131614f) (listing deployments in `mirrord ls` is not yet released, so no deployments in the dialog)

### CODE REVIEW

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt [hunk 1]
```diff
@@ -27,59 +30,130 @@ object MirrordExecDialog {
...
-            if (idx != -1) {
-                targets.removeAt(idx)
-                targets.add(0, it)
+    fun selectTargetDialog(availableTargets: List<String>): String? {
+        val targetsState = object {
+            private var _targets = availableTargets
+                    .sorted()
+                    .toMutableList()
+                    .apply {
+                        // Initially, the last used target is at the top
```
Comment (infiniteregrets) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt:39]: imho, this should be a class called `TargetState` and you can just do `val targetsState = TargetsState(availableTargets)`

**Review by infiniteregrets — COMMENTED** [review]
looks great overall. look at the comment (:

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt [hunk 2]
```diff
@@ -27,59 +30,130 @@ object MirrordExecDialog {
...
+                get() {
+                    return this._targets + targetlessTargetName
+                }
+            var pods = true
+                set(value) {
+                    field = value
+                    this.update()
+                }
+            var deployments = true
+                set(value) {
```
Comment (eyalb181) [inline modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt:59]: Are these persistent? Does it remember what categories I had checked the next time I run mirrord?

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt (hunk 1 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt]: Done, thanks!

File: modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt (hunk 2 above)
Comment (Razz4780) [reply modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecDialog.kt]: Now they are :)

**Comment by Razz4780** [comment]
Minor change - last selected target is now **always** at the top if it's not filtered out

**Review by infiniteregrets — APPROVED** [review]

**Pull request merged**